components/server/src/workspace/context-service.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { WorkspaceDB, DBWithTracing, TracedWorkspaceDB } from "@gitpod/gitpod-db/lib";
+import {
+    CommitContext,
+    PrebuiltWorkspace,
+    PrebuiltWorkspaceContext,
+    User,
+    WorkspaceContext,
+    Project,
+    SnapshotContext,
+} from "@gitpod/gitpod-protocol";
+import { ApplicationError, ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { inject, injectable } from "inversify";
+import { ContextParser } from "./context-parser-service";
+import { ConfigProvider } from "./config-provider";
+import { ProjectsService } from "../projects/projects-service";
+import { OpenPrebuildContext, WithDefaultConfig } from "@gitpod/gitpod-protocol/lib/protocol";
+import { IncrementalWorkspaceService } from "../prebuilds/incremental-workspace-service";
+import { Authorizer } from "../authorization/authorizer";
+
+@injectable()
+export class ContextService {
```

Comment:
ideally we should add DB tests for it

Did we only move code as is? Or there was change of logic?